CRM/Core/BAO/CustomGroup.php

```diff
@@ -1377,7 +1377,13 @@ public static function setDefaults(&$groupTree, &$defaults, $viewMode = FALSE, $
                 }
               }
               else {
-                $checkedValue = explode(CRM_Core_DAO::VALUE_SEPARATOR, substr($value, 1, -1));
+                if (is_array($value) && count($value)) {
```

Comment:
I have tested this patch by adding a new contribution & filling it in with contribution details & custom field details but no amount. I save the custom data disappears. However, with this patch it does not.

I'm just going through the patch to see if it makes sense / seems risk free. 

This line seems risk free as an array would not have previously been handled & the non array path is unchanged